File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    const alphabets = ['A', 'E', 'I', 'O', 'U'];
    let count = 0;
    let answer = 0;
    
    function dfs(str) {
        if (str === word) {
            answer = count;
            return;
        }
        if (str.length === 5) return;
        
        for (let i = 0; i < 5; i++) {
            count++;
            dfs(str + alphabets[i]);
            if (answer) return;
        }
    }
    
    dfs("");
    
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    vowels = ['A', 'E', 'I', 'O', 'U']
    ans = [0]
    cnt = [0]
    
    def dfs(str):
        if str == word:
            ans[0] = cnt[0]
            return
        
        if len(str) == 5:
            return
        
        for i, vowel in enumerate(vowels):
            cnt[0] += 1
            dfs(str + vowels[i])
            if ans[0]:
                return
            
    dfs('')
    
    return ans[0]